run.py

```diff
@@ -122,36 +125,32 @@ def config_validator():
-                        print 'backup db folder does not exist'
-                        os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
-                    else:
-                        print 'backup db folder exists'
-                        # Retain last 3 recent backup files
-                        no_of_backup_files = len(os.listdir(app.config['BACKUP_DEFAULT_DEST']))
-                        if no_of_backup_files > 3:
-                            for idx, val in enumerate(os.listdir(app.config['BACKUP_DEFAULT_DEST'])):
-                                if idx < no_of_backup_files - 2:
-                                    os.remove(app.config['BACKUP_DEFAULT_DEST'] + val)
-                    current_ts = time.time()
-                    time_stamp = datetime.datetime.fromtimestamp(current_ts).strftime('%Y-%m-%d-%H:%M:%S')
-                    issue_backup_db = 'issues_backup_' + time_stamp + '.db'
-                    os.rename(os.getcwd() + '/issues.db', app.config['BACKUP_DEFAULT_DEST'] + issue_backup_db)
-                    try:
-                        os.remove(os.path.join(os.getcwd(), 'issues.db'))
-                    except OSError:
-                        print 'Backup issues.db complete'
-
-            except IOError:
-                sys.exit(FILE_NOT_FOUND_ERROR)
-
+            issue_engine = create_engine('sqlite:///' + os.path.join(
+                app.config['BASE_DIR'], 'issues.db'))
+            issue_session_maker = sessionmaker(autocommit=False,
+                                               autoflush=False,
+                                               bind=issue_engine)
+            issue_db = scoped_session(issue_session_maker)
+            # Take a backup if issues.db has data dump.
+            if issue_db().execute('select count(*) from webcompat_issues').fetchall()[0][0] > 0:
+                if not os.path.exists(app.config['BACKUP_DEFAULT_DEST']):
+                    print 'Backup db folder does not exist'
+                    os.makedirs(app.config['BACKUP_DEFAULT_DEST'])
+                else:
+                    print 'Backup db folder exists'
+                current_ts = time.time()
+                time_stamp = datetime.datetime.fromtimestamp(current_ts).strftime('%Y-%m-%d-%H:%M:%S')
+                issue_backup_db = 'issues_backup_' + time_stamp + '.db'
+                os.rename(os.getcwd() + '/issues.db', app.config['BACKUP_DEFAULT_DEST'] + issue_backup_db)
+                # Retain last 3 recent backup files
+                no_of_backup_files = len(os.listdir(app.config['BACKUP_DEFAULT_DEST']))
+                print no_of_backup_files
+                if no_of_backup_files > 3:
+                    for idx, val in enumerate(os.listdir(app.config['BACKUP_DEFAULT_DEST'])):
+                        print idx
+                        if idx < no_of_backup_files - 3:
+                            os.remove(app.config['BACKUP_DEFAULT_DEST'] + val)
```

Comment:
ok let's see a better way to do that. Let's say we have this list of files.

```
→ ls -1
db2016-08-17T01-03-20
db2016-08-17T05-02-20
db2016-08-18T01-03-20
db2016-08-18T04-05-06
db2016-08-18T14-03-01
db2016-08-18T14-03-20
```

And the list is: 
```python
>>> import os
>>> files_list = os.listdir('.')
>>> # And let's say the files_list comes into a different order
>>> files_list
['db2016-08-17T01-03-20', 'db2016-08-18T14-03-01', 'db2016-08-18T14-03-20', 'db2016-08-17T05-02-20', 'db2016-08-18T01-03-20', 'db2016-08-18T04-05-06']
>>> # Let's do a sort so the files will now be in alphabetical order
>>> files_list.sort()
>>> files_list
['db2016-08-17T01-03-20', 'db2016-08-17T05-02-20', 'db2016-08-18T01-03-20', 'db2016-08-18T04-05-06', 'db2016-08-18T14-03-01', 'db2016-08-18T14-03-20']
>>> # the oldest are at the beginning and the most recent ones at the end of the list
>>> # we want to keep the most recent one. So we need the list of files to remove.
>>> # This will cut the list by removing the 3 last items in the list
>>> files_list[:3]
['db2016-08-17T01-03-20', 'db2016-08-17T05-02-20', 'db2016-08-18T01-03-20']
>>> # so basically you can do 
>>> for old_file in files_list[:3]:
...     os.remove(old_file)
... 
```

Let's see in our directory:

```
→ ls -1
db2016-08-18T04-05-06
db2016-08-18T14-03-01
db2016-08-18T14-03-20
```